config/history.go

```diff
@@ -220,35 +248,26 @@ func AddClipboardItem(text, fp string) error {
 	data.ClipboardHistory = append([]ClipboardItem{item}, data.ClipboardHistory...)
 
 	if len(data.ClipboardHistory) > ClipseConfig.MaxHistory {
-		for i := len(data.ClipboardHistory) - 1; i >= 0; i-- { // remove the first unpinned entry starting with the oldest
+		for i := len(data.ClipboardHistory) - 1; i >= 0; i-- {
+			// remove the first unpinned entry starting with the oldest
 			if !data.ClipboardHistory[i].Pinned {
 				data.ClipboardHistory = append(data.ClipboardHistory[:i], data.ClipboardHistory[i+1:]...)
 				break
 			}
 		}
 	}
 
-	if err = saveDataToFile(data); err != nil {
+	if err := WriteUpdate(data); err != nil {
 		return err
 	}
 	return nil
 }
 
 // This pins and unpins an item in the clipboard
 func TogglePinClipboardItem(timeStamp string) (bool, error) {
-	var data ClipboardHistory
+	var data ClipboardHistory = fileContents()
 	var pinned bool // gets the pinned state of the iteem
 
-	fileData, err := os.ReadFile(ClipseConfig.HistoryFilePath)
-	if err != nil {
-		return pinned, err
-	}
-
-	err = json.Unmarshal(fileData, &data)
-	if err != nil {
-		return pinned, err
-	}
-
 	for i := range data.ClipboardHistory {
```

Comment:
Wait

Why this?

```
         for i := range data.ClipboardHistory { 
                  if data.ClipboardHistory[i].Recorded == timeStamp {
```

And not this ?

```suggestion
 	for _, ch := range data.ClipboardHistory {
       if ch.Recorded == timeStamp
```

Reply:
Also agree. Will get this and the above updated.

Reply:
Oh, I just remembered the reason for this.. it saves you from having to re-pack an array with new values to then write to the history file. Eg, the alternative would mean something like:
```go
var updatedClipboardHistory []ClipboardItem
	for _, item := range data.ClipboardHistory {
		if item.Recorded == timeStamp {
			pinned = item.Pinned
			item.Pinned = !item.Pinned
		}
		updatedClipboardHistory = append(updatedClipboardHistory, item)
	}
``` 
This doesn't feel better, but maybe I'm missing something?

Reply:
OK, indeed

I missed you reassigned it

Then it should be something like this (pseudo code)

```
var updatedClipboardHistory []ClipboardItem

```go
for i, item := range data.ClipboardHistory { 	
	if item.Recorded == timeStamp {
      pinned = item.Pinned
      item.Pinned = !item.Pinned
      data.ClipboardHistory[i] = item
      break
}
```

Reply:
I edited my previous comment if you saw my first version

Reply:
@ccoVeille Yess thanks I like that a lot more. Have been a bit hasty merging things but but create a new pr over [here](https://github.com/savedra1/clipse/pull/73/commits/a08ba2d7f5704509c41e212ea877f7e31b23b8cd)